src/webgl/p5.Camera.js

```diff
@@ -1638,6 +1638,324 @@ p5.Camera = class Camera {
+
+    // Prepare each of rotation matrix from their camera matrix
+    const mat0 = cam0.cameraMatrix.mat4;
+    const mat1 = cam1.cameraMatrix.mat4;
+    const m0 = [
+      mat0[0], mat0[1], mat0[2],
+      mat0[4], mat0[5], mat0[6],
+      mat0[8], mat0[9], mat0[10]
+    ];
+    const m1 = [
+      mat1[0], mat1[1], mat1[2],
+      mat1[4], mat1[5], mat1[6],
+      mat1[8], mat1[9], mat1[10]
+    ];
+    // Create the inverse matrix of mat0 by transposing mat0,
+    // and multiply it to mat1 from the right.
+    // This matrix represents the difference between the two.
+    const m = [
+      m1[0] * m0[0] + m1[1] * m0[1] + m1[2] * m0[2],
+      m1[0] * m0[3] + m1[1] * m0[4] + m1[2] * m0[5],
+      m1[0] * m0[6] + m1[1] * m0[7] + m1[2] * m0[8],
+
+      m1[3] * m0[0] + m1[4] * m0[1] + m1[5] * m0[2],
+      m1[3] * m0[3] + m1[4] * m0[4] + m1[5] * m0[5],
+      m1[3] * m0[6] + m1[4] * m0[7] + m1[5] * m0[8],
+
+      m1[6] * m0[0] + m1[7] * m0[1] + m1[8] * m0[2],
+      m1[6] * m0[3] + m1[7] * m0[4] + m1[8] * m0[5],
+      m1[6] * m0[6] + m1[7] * m0[7] + m1[8] * m0[8]
+    ];
+
+    // Calculate the trace and from it the cos value of the angle.
+    // An orthogonal matrix is just an orthonormal basis. If this is not the identity
+    // matrix, it is a centered orthonormal basis plus some angle of rotation about
+    // some axis. That's the angle. Letting this be theta, trace becomes 1+2cos(theta).
+    let cosTheta = 0.5 * (m[0] + m[4] + m[8] - 1);
+    // If the angle is close to 0, the two matrices are very close,
+    // so in that case we execute linearly interpolate.
+    if (1 - cosTheta < 0.0000001) {
+      // Obtain the front vector and up vector by linear interpolation
+      // and normalize them.
+      const lerpedFront = new p5.Vector(
+        (1 - amt) * m0[2] + amt * m1[2],
+        (1 - amt) * m0[5] + amt * m1[5],
+        (1 - amt) * m0[8] + amt * m1[8]
+      ).normalize();
+      const lerpedUp = new p5.Vector(
+        (1 - amt) * m0[1] + amt * m1[1],
+        (1 - amt) * m0[4] + amt * m1[4],
+        (1 - amt) * m0[7] + amt * m1[7]
+      ).normalize();
+
+      // set the camera
+      // The eye position and center position are calculated based on the front vector.
+      this.camera(
+        lerpedMediumX + ratio * lerpedDist * lerpedFront.x,
+        lerpedMediumY + ratio * lerpedDist * lerpedFront.y,
+        lerpedMediumZ + ratio * lerpedDist * lerpedFront.z,
+        lerpedMediumX + (ratio-1) * lerpedDist * lerpedFront.x,
+        lerpedMediumY + (ratio-1) * lerpedDist * lerpedFront.y,
+        lerpedMediumZ + (ratio-1) * lerpedDist * lerpedFront.z,
+        lerpedUp.x, lerpedUp.y, lerpedUp.z
+      );
+      return;
+    }
+
+    // Calculates the axis vector and the angle of the difference orthogonal matrix.
+    // The axis vector is what I explained earlier in the comments.
+    // similar calculation is here:
+    // https://github.com/mrdoob/three.js/blob/dev/src/math/Quaternion.js#L294
```

Comment:
Can we use the permalink URL to this line instead? (Obtained through the three dot menu on github when you select a line, I think it just uses the commit hash in the URL instead of the branch, in case this file changes on the dev branch in the future.) https://github.com/mrdoob/three.js/blob/883249620049d1632e8791732808fefd1a98c871/src/math/Quaternion.js#L294

Reply:
Ok, I'll fix that.